### DISCUSSION TITLE
Calling trigger on a LazyQuery always starts new request

### DISCUSSION THREAD

**Discussion by kjg** [reduxjs/redux-toolkit#1279] [Q&A] [answered]
It appears that when I use the trigger function returned from a LazyQuery, it always initiates a new request to the server even if there is cached data.

Is this to be expected? I would expect the trigger function to return cached data when called with the same args as a previous trigger. If not, how do I access the cached data?

**Comment by phryneas** [comment, marked as answer]
Huh, this is actually undocumented and... not even in the TS types?

To answer your question, the *real* signature of the `trigger` function is
```ts
function (arg: any, preferCacheValue = false) {
```
so you can decide if it should use the cache with the second argument.

We need to get this in the docs and types though. Thanks for pointing it out.

**Reply by kjg** [reply]
Thanks, this does the trick! I'm also working on a Load More situation like in the [Infinite scroll discussion](https://github.com/reduxjs/redux-toolkit/discussions/1163#discussioncomment-855577).

This library has been such a huge help and saved me so much time, thanks so much for it!

**Comment by akashshyamdev** [comment]
@phryneas I'd like to contribute to this. I'm new to this repo so would appreciate some guidance. According to the changes highlighted by you and @kjg, I think the changes should be made in this section:

1. Doc updates - add a new note at the bottom(before `useLazyQuerySubscription` heading), saying "People expect the trigger function to return cached data when called with the same args as a previous trigger. However, it always initiates a new request to the server even if there is cached data."

![Screenshot 2021-09-16 at 19 50 26](https://user-images.githubusercontent.com/56759828/133629077-1c601239-01e7-4f6c-8cda-b7a2812fa0b2.png)

2. Type definitions - based on the short discussion here, it seems that the `preferCacheValue` argument is not present in the type definitions. Could you point me out to the type definition files?

Thanks a lot for reading, looking forward to your reply!

**Comment by phryneas** [comment]
You can find that type here. Adding to the comment above it will also end up in the documentation.

https://github.com/reduxjs/redux-toolkit/blob/9e772d2bb9434755af0142ee276954338a990e98/packages/toolkit/src/query/react/buildHooks.ts#L183-L191

Right now, there are two options to handle this:

* just add the type annotation and a note that this will trigger by default, but can be deactivated

* handle this as a new option that is also breaking old behavior. 
  I would actually prefer doing this (defaulting to `preferCacheValue=true`), but I think it would be too breaking for a minor release and I don't want to wait with this until we release a 2.0. What do you think @markerikson? Could we do something like that in a minor or is that stretching it too far?

**Reply by markerikson** [reply]
If we feel that the current behavior is buggy, then it's legit fodder for a minor.

(I suspect not that many people are using this API so far anyway.)

**Reply by akashshyamdev** [reply]
Thanks for the advice regarding the type @phryneas. What do you mean by this "Adding to the comment above it will also end up in the documentation"? Also, should I create a PR for the above?

**Reply by markerikson** [reply]
@akashshyamdev Several portions of the docs site are automatically generated based on docblock comments in the codebase.  And yes, file a PR.

**Reply by akashshyamdev** [reply]
@markerikson I've tried updating the code blocks but the changes are not being reflected in the docs. The PR is [here] (https://github.com/reduxjs/redux-toolkit/pull/1541)

**Reply by akashshyamdev** [reply]
@phryneas and @markerikson I've filed a pr regarding this

**Reply by phryneas** [reply]
Yes, I'm sorry, I'm a little swamped - but don't worry, it is not forgotten, it might just take a while :)

**Comment by akashshyamdev** [comment]
@markerikson and @phryneas I've filed the PR 👍

**Comment by gabrielsanttana** [comment]
Hey guys, just came up with the issue here. I would like to know the current status of this thread, and if there's something to do about it. As I've seen, the doc has been updated and it now has a note about the `preferCacheValue` argument, but it's still not available for the trigger from the `useLazyQuery` hook.

![image](https://user-images.githubusercontent.com/47339825/142475183-cbcf4c34-f97b-43a1-b2a9-234113e5d14b.png)

**Reply by phryneas** [reply]
It should be *available*, but it's not in the TS types of 1.6.x yet. You can try out 1.7 beta which should have it.

**Reply by gabrielsanttana** [reply]
Oh, yeah. I've just tried it out and it worked. I'll try out version 1.7. Thank you for the response, @phryneas!

**Comment by bryanprimus** [comment]
not sure why i still not get the types in the latest version

**Reply by bryanprimus** [reply]
do i need to also upgrade my typescript or something ?

**Reply by phryneas** [reply]
As per #3072, your types are perfectly fine, you just had some incorrect assumptions on how to use this. It's `trigger(arg, true)`, not `trigger(arg, { preferCacheValue: true })`. The types correctly reflect that.

**Reply by bryanprimus** [reply]
is the default for `preferCacheValue` still `false`? or is it already updated to `true`

**Reply by phryneas** [reply]
That would be a breaking change, so the soonest we do that would be 2.0
